src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -515,8 +515,12 @@ def get_timesteps(self, num_inference_steps, strength, device, denoising_start=N
                     - (denoising_start * self.scheduler.config.num_train_timesteps)
                 )
             )
-            timesteps = list(filter(lambda ts: ts < discrete_timestep_cutoff, timesteps))
-            return torch.tensor(timesteps), len(timesteps)
+
+            num_inference_steps = (timesteps < discrete_timestep_cutoff).sum()
+            if self.scheduler.order == 2:
+                num_inference_steps = num_inference_steps + 1
```

Comment:
```suggestion
                num_inference_steps = num_inference_steps + 1
                
                # because t_n+1 >= t_n, we slice the timesteps starting from the end
```